server/swap/swap.go

```diff
@@ -200,13 +204,26 @@ func NewSwapper(cfg *Config) *Swapper {
 	}
 	// The swapper is only concerned with two types of client-originating
 	// method requests.
-	authMgr.RegisterMethod(msgjson.InitRoute, swapper.handleInit)
-	authMgr.RegisterMethod(msgjson.RedeemRoute, swapper.handleRedeem)
+	authMgr.Route(msgjson.InitRoute, swapper.handleInit)
+	authMgr.Route(msgjson.RedeemRoute, swapper.handleRedeem)
 
+	swapper.wg.Add(1)
 	go swapper.start()
 	return swapper
 }
 
+// Stop begins the Swapper's shutdown. Use WaitForShutdown after Stop to wait
+// for shutdown to complete.
+func (s *Swapper) Stop() {
+	log.Infof("Swapper shutting down...")
+	s.cancel()
+}
+
+// WaitForShutdown waits until the main swapper loop is finished.
+func (s *Swapper) WaitForShutdown() {
+	s.wg.Wait()
+}
+
```

Comment:
Note: Still need a `TxMonitored` method for `MarketTunnel`, which is probably best implemented as a method on `Swapper`. Probably out of scope for this PR. I haven't checked if you had a solution in #81 yet though.

Reply:
No, #81 doesn't go very far yet.

Aside from `SubmitOrder`, `MarketTunnel` is not implemented by `Market` in this PR.  There is a fair bit of additional work on Market to track the locked outpoints/coinIDs.  I'm planning to follow up on this next since there are a number of design questions around `OutpointLocked`, `Cancelable`, and `TxMonitored` that does directly related to the epoch processing bits.  I would agree that `TxMonitored` would be essentially forwarded to `Swapper`.